Script consolidation

mkolopanis opened the pull request:
Removes the redundant script lwa_bifrost_DFT.py  and adds the DFT functionality to the main script. 
A bit of formatting for readability.
Added the synchronization for the DFT accumulation. closes #13 

I removed the TransposeOp. It served no purpose anymore since the transpose appears to happen upstream? It was no longer in the original script and with the changes it was not needed for the DFT one.

See comparison with master in this notebook. The DFT shouldn't match but does now between multiple runs.

[script_consolidate.pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5234622/script_consolidate.pdf)
multiple runs of the dft code on the main branch
[dft_comparison.pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5234624/dft_comparison.pdf)
multiple runs of dft code with this branch.
[dft_comparison_update.pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5234625/dft_comparison_update.pdf)

mkolopanis committed:
a few formatting changes when comaring two LWA_bifrost scripts

mkolopanis committed:
try to patch through changes from lwa_bifrost into dft script

mkolopanis committed:
remove TransposeOP, no logger needed because data is already transposed

mkolopanis committed:
remove extra script, add DFT code to original

mkolopanis committed:
remove some unused variables

mkolopanis commented:
@KentJames Do you want to take a look to make sure I got the DFT code moved over correctly?

mkolopanis committed:
force nstand to be an int, force int division for ishape and oshape in rings

mkolopanis commented:
Here's another notebook comparison. I ran the code in a python 3 environment (the 'py3_test' file). 
[script_consolidate (2).pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5336284/script_consolidate.2.pdf)

KentJames commented:
Sorry I clearly missed this. I’ll take a look. 


> On 18 Sep 2020, at 04:20, Matthew Kolopanis <notifications@github.com> wrote:
> 
> ﻿
> @KentJames Do you want to take a look to make sure I got the DFT code moved over correctly?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

adampbeardsley commented on LWA/LWA_bifrost.py:
```diff
@@ -1150,6 +1249,343 @@ def main(self):
+                    for j in numpy.arange(lm_matrix.shape[0]):
+                        lm_matrix[i, j] = numpy.asarray([i * lm_step - 1.0, j * lm_step - 1.0, 0.0])
+                lm_vector = lm_matrix.reshape((self.skymodes, 3))
+                print("shapes: locs {locs}, phases {phases}".format(
+                    locs=locs.shape, phases=phases.shape,
+                ))
```
I think this print statement is new. Do we want it?

adampbeardsley reviewed (commented):
I've looked it over and the only comment I have is that there is a new print statement. I think it's fine to leave it in, unless it was a diagnostic that we don't normally need.
I'll hold off on "approving" in case James wants to have a look.
Nice work Matt!

mkolopanis commented:
I think it is new as well probably want to remove it. It most likely was a
put in to help figure out some int vs float issues I had.

On Wed, Oct 7, 2020, 11:15 Adam Beardsley <notifications@github.com> wrote:

> *@adampbeardsley* commented on this pull request.
>
> I've looked it over and the only comment I have is that there is a new
> print statement. I think it's fine to leave it in, unless it was a
> diagnostic that we don't normally need.
> I'll hold off on "approving" in case James wants to have a look.
> Nice work Matt!
> ------------------------------
>
> In LWA/LWA_bifrost.py
> <https://github.com/epic-astronomy/LWA_EPIC/pull/14#discussion_r501211316>
> :
>
> > +                print("shapes: locs {locs}, phases {phases}".format(
> +                    locs=locs.shape, phases=phases.shape,
> +                ))
>
> I think this print statement is new. Do we want it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/epic-astronomy/LWA_EPIC/pull/14#pullrequestreview-504140329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACB7BLOOK5XKKJ4CJKSFZJTSJSV4LANCNFSM4RPKM2TA>
> .
>

KentJames commented:
Everything looks kosher to me. And thanks for going through and commenting everything properly Matt... clearly that fell by the wayside a bit in my rush to write up. 

I've been thinking for a while about how we can use these both together.. EPIC for the dense core of antennas, and then EPIC-DFT for any outrigger antennas.. seems like we aren't far off now you've merged the two strands into one whole.

mkolopanis committed:
remove print statement

mkolopanis committed:
numpy-ize some nested for loops

mkolopanis commented:
I think I added comments as I figured out what things did :smile: it was educational for me.


Here's a notebook showing the vectorized version of the code I changed produces identical outputs.
[numpize_for_loops.pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5349435/numpize_for_loops.pdf)

And that the images are identical 
[dft_comparison (1).pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5349454/dft_comparison.1.pdf)

also this shows images are identical in python 2 and 3 now 
[script_consolidate (3).pdf](https://github.com/epic-astronomy/LWA_EPIC/files/5349469/script_consolidate.3.pdf)

To be fair in all the comparison I only opened a single file and looked at all the images it contained. Could be more thorough and compare across all frequencies if there is a desire.

mkolopanis commented:
@adampbeardsley do you want to take another look over this after the DFT blessing was bestowed?
